### PR TITLE
daemon: cache error on policy import for CNP

### LINKED ISSUE #5211: Enforcing flag set to true after annotations were updated for a non-valid policy
steps to reproduce:

1. write this into `cidr.yaml`
```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
spec:
  endpointSelector:
    matchLabels:
      app: myService
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
```

2. write this into `cidr-with-annotations.yaml`
```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
  annotations:
    foo: bar
spec:
  endpointSelector:
    matchLabels:
      app: myService
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
```

3. `kubectl create -f ./cidr.yaml`

4. Verify node status

```
$ kubectl get cnp cidr-rule  -o yaml
apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
metadata:
  creationTimestamp: 2018-08-15T09:57:14Z
  generation: 1
  name: cidr-rule
  namespace: default
  resourceVersion: "188294"
  selfLink: /apis/cilium.io/v2/namespaces/default/ciliumnetworkpolicies/cidr-rule
  uid: 96304927-a071-11e8-bab3-0800278b5835
spec:
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
  endpointSelector:
    matchLabels:
      app: myService
status:
  nodes:
    k8s1:
      error: 'Adding specified prefixes would result in too many prefix lengths (current:
        3, result: 25, max: 18)'
      lastUpdated: 2018-08-15T09:57:15.668629552Z
    k8s2:
      error: 'Adding specified prefixes would result in too many prefix lengths (current:
        3, result: 25, max: 18)'
      lastUpdated: 2018-08-15T09:57:14.654258673Z
```

5. Apply policy with annotations `kubectl apply -f ./cidr-with-annotations.yaml`

6. Verify node status. 

Notes:
- `k8s1` is running against master
- `k8s2` is running without https://github.com/cilium/cilium/pull/5192

```
$ kubectl get cnp cidr-rule  -o yaml
apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
metadata:
  annotations:
    foo: bar
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{"foo":"bar"},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
  creationTimestamp: 2018-08-15T09:57:14Z
  generation: 1
  name: cidr-rule
  namespace: default
  resourceVersion: "188365"
  selfLink: /apis/cilium.io/v2/namespaces/default/ciliumnetworkpolicies/cidr-rule
  uid: 96304927-a071-11e8-bab3-0800278b5835
spec:
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
  endpointSelector:
    matchLabels:
      app: myService
status:
  nodes:
    k8s1:
      annotations:
        foo: bar
        kubectl.kubernetes.io/last-applied-configuration: |
          {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{"foo":"bar"},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
      enforcing: true
      lastUpdated: 2018-08-15T09:57:55.518004599Z
      ok: true
    k8s2:
      annotations:
        foo: bar
        kubectl.kubernetes.io/last-applied-configuration: |
          {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{"foo":"bar"},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
      error: 'Adding specified prefixes would result in too many prefix lengths (current:
        3, result: 25, max: 18)'
      lastUpdated: 2018-08-15T09:57:56.56775988Z
```

### PR DESCRIPTION
**Pull request by ianvernon**
If an error occurs policy importing, it should be cached so that if the
annotations are updated for a rule, they are aware of said import error, and can
access the error from a map of CiliumNetworkPolicy to policy-import related
metadata (revision number, error from import), and pass this appropriately to
the function to update the status of the CNP. This ensures that annotations
updates do not override the "Error" field of the CNP NodeStatus if there
was an error importing the policy.

Fixes: 43e3c7ee3b0e ("daemon: always use same sync func for CNPNS")

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #5211 

```release-note
Do not override error from policy import if only annotations are updated for rule which failed to be imported
```

I followed the instructions from #5211 and confirmed that that fixed the issue:

1. Create file `cidr.yaml`:

```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
spec:
  endpointSelector:
    matchLabels:
      app: myService
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
```

2. Apply this policy:

```
$ kubectl apply -f cidr.yaml 
ciliumnetworkpolicy.cilium.io/cidr-rule created
```

3. Check that the error field says that policy is not valid in the CNP NodeStatus:
```
$ kubectl get cnp cidr-rule  -o yaml
apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
  creationTimestamp: 2018-08-15T16:05:27Z
  generation: 1
  name: cidr-rule
  namespace: default
  resourceVersion: "10230"
  selfLink: /apis/cilium.io/v2/namespaces/default/ciliumnetworkpolicies/cidr-rule
  uid: 068df04a-a0a5-11e8-a53e-0800278b5835
spec:
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
  endpointSelector:
    matchLabels:
      app: myService
status:
  nodes:
    k8s1:
      annotations:
        kubectl.kubernetes.io/last-applied-configuration: |
          {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
      error: 'Adding specified prefixes would result in too many prefix lengths (current:
        3, result: 25, max: 18)'
      lastUpdated: 2018-08-15T16:05:27.492763941Z
```

4. Store the following iteration of the above policy in a file `cidr-2.yaml`, which only contains updates to annotations:
```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
  annotations:
    foo: bar
spec:
  endpointSelector:
    matchLabels:
      app: myService
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
```

5. Apply said policy:

```
$ kubectl apply -f cidr-2.yaml 
ciliumnetworkpolicy.cilium.io/cidr-rule configured
```

6. Check that the error field is still populated with the same error as above, but that the annotations are updated:
```
$ kubectl get cnp cidr-rule  -o yaml
apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
metadata:
  annotations:
    foo: bar
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{"foo":"bar"},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
  creationTimestamp: 2018-08-15T16:05:27Z
  generation: 1
  name: cidr-rule
  namespace: default
  resourceVersion: "10279"
  selfLink: /apis/cilium.io/v2/namespaces/default/ciliumnetworkpolicies/cidr-rule
  uid: 068df04a-a0a5-11e8-a53e-0800278b5835
spec:
  egress:
  - toCIDR:
    - 20.1.1.1/32
    - 20.2.1.1/31
    - 20.3.1.1/30
    - 20.4.1.1/29
    - 20.5.1.1/28
    - 20.6.1.1/27
    - 20.7.1.1/26
    - 20.8.1.1/25
    - 20.9.1.1/24
    - 20.10.1.1/23
    - 20.11.1.1/22
    - 20.12.1.1/21
    - 20.13.1.1/20
    - 20.14.1.1/19
    - 20.15.1.1/18
    - 20.16.1.1/17
    - 20.17.1.1/16
    - 20.18.1.1/15
    - 20.19.1.1/14
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
  endpointSelector:
    matchLabels:
      app: myService
status:
  nodes:
    k8s1:
      annotations:
        foo: bar
        kubectl.kubernetes.io/last-applied-configuration: |
          {"apiVersion":"cilium.io/v2","kind":"CiliumNetworkPolicy","metadata":{"annotations":{"foo":"bar"},"name":"cidr-rule","namespace":"default"},"spec":{"egress":[{"toCIDR":["20.1.1.1/32","20.2.1.1/31","20.3.1.1/30","20.4.1.1/29","20.5.1.1/28","20.6.1.1/27","20.7.1.1/26","20.8.1.1/25","20.9.1.1/24","20.10.1.1/23","20.11.1.1/22","20.12.1.1/21","20.13.1.1/20","20.14.1.1/19","20.15.1.1/18","20.16.1.1/17","20.17.1.1/16","20.18.1.1/15","20.19.1.1/14"]},{"toCIDRSet":[{"cidr":"10.0.0.0/8","except":["10.96.0.0/12"]}]}],"endpointSelector":{"matchLabels":{"app":"myService"}}}}
      error: 'Adding specified prefixes would result in too many prefix lengths (current:
        3, result: 25, max: 18)'
      lastUpdated: 2018-08-15T16:06:06.389310457Z
```

This confirms that this indeed fixes #5211 .

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5215)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by joestringer** [comment]
I don't think that the `release-note/bug` annotation is required since this fixes an issue only present in master.

File: daemon/k8s_watcher.go [hunk 1]
```diff
@@ -93,50 +93,59 @@ var (
...
+	policyImportError error
 }
 
-func (r *ruleRevisionCache) upsert(cnp *cilium_v2.CiliumNetworkPolicy, revision uint64) {
+func initRuleMetadataCache() {
+	importMetadataCache.ruleImportMetadataMap = make(map[string]policyImportMetadata)
+}
+
+func (r *ruleImportMetadataCache) upsert(cnp *cilium_v2.CiliumNetworkPolicy, revision uint64, importErr error) {
 	r.mutex.Lock()
```
Comment (aanm) [inline daemon/k8s_watcher.go]: create the `policyImportMetadata` object before the Lock and put the lock/unlock only on the write of the map

File: daemon/k8s_watcher.go [hunk 2]
```diff
@@ -93,50 +93,59 @@ var (
...
-func (r *ruleRevisionCache) delete(cnp *cilium_v2.CiliumNetworkPolicy) {
+func (r *ruleImportMetadataCache) delete(cnp *cilium_v2.CiliumNetworkPolicy) {
 	r.mutex.Lock()
 	defer r.mutex.Unlock()
 	if cnp == nil {
 		return
 	}
 	podNSName := k8sUtils.GetObjNamespaceName(&cnp.ObjectMeta)
-	delete(r.ruleRevisionMapping, podNSName)
+	delete(r.ruleImportMetadataMap, podNSName)
```
Comment (aanm) [inline daemon/k8s_watcher.go]: the lock/unlock only needs wrap this delete

File: daemon/k8s_watcher.go [hunk 3]
```diff
@@ -93,50 +93,59 @@ var (
...
 	r.mutex.RLock()
 	defer r.mutex.RUnlock()
 	if cnp == nil {
-		return 0, false
+		return policyImportMetadata{}, false
 	}
 	podNSName := k8sUtils.GetObjNamespaceName(&cnp.ObjectMeta)
-	revision, ok := r.ruleRevisionMapping[podNSName]
-	return revision, ok
+	policyImportMeta, ok := r.ruleImportMetadataMap[podNSName]
```
Comment (aanm) [inline daemon/k8s_watcher.go]: the RLock/RUnlock only needs to wrap this line

File: daemon/k8s_watcher.go [hunk 4]
```diff
@@ -93,50 +93,59 @@ var (
 
 	k8sCM = controller.NewManager()
 
-	ruleRevCache ruleRevisionCache
```
Comment (aanm) [inline daemon/k8s_watcher.go:96]: [1] here

File: daemon/k8s_watcher.go [hunk 5]
```diff
@@ -178,7 +187,7 @@ func init() {
 		k8sErrorHandler,
 	}
 
-	initRuleRevCache()
```
Comment (aanm) [inline daemon/k8s_watcher.go]: This function can be removed and the make of the map can be directly done in [1]

File: daemon/k8s_watcher.go [hunk 6]
```diff
@@ -93,50 +93,59 @@ var (
...
+
+func (r *ruleImportMetadataCache) upsert(cnp *cilium_v2.CiliumNetworkPolicy, revision uint64, importErr error) {
 	r.mutex.Lock()
 	defer r.mutex.Unlock()
 	if cnp == nil {
 		return
 	}
 	podNSName := k8sUtils.GetObjNamespaceName(&cnp.ObjectMeta)
-	r.ruleRevisionMapping[podNSName] = revision
+	r.ruleImportMetadataMap[podNSName] = policyImportMetadata{
```
Comment (aanm) [inline daemon/k8s_watcher.go]: Did you consider storing the CNPNS itself instead of creating a new structure to hold this data?

File: daemon/k8s_watcher.go [hunk 7]
```diff
@@ -1507,15 +1516,15 @@ func (d *Daemon) updateCiliumNetworkPolicyV2AnnotationsOnly(ciliumV2Store cache.
...
 	// to be updated by addCiliumNetworkPolicyV2 before calls to
 	// updateCiliumNetworkPolicyV2 are invoked.
-	rev, _ := ruleRevCache.get(cnp)
+	policyImportMetadata, _ := importMetadataCache.get(cnp)
 
 	k8sCM.UpdateController(ctrlName,
 		controller.ControllerParams{
 			DoFunc: func() error {
-				return cnpNodeStatusController(ciliumV2Store, cnp, rev, scopedLog, nil)
+				return cnpNodeStatusController(ciliumV2Store, cnp, policyImportMetadata.revision, scopedLog, policyImportMetadata.policyImportError)
```
Comment (aanm) [inline daemon/k8s_watcher.go]: consider renaming `policyImportMetadata` to `metadata` as this line already has 149 columns

**Review by aanm — DISMISSED** [review]

File: daemon/k8s_watcher.go (hunk 6 above)
Comment (ianvernon) [reply daemon/k8s_watcher.go]: I didn't want to do that for multiple reasons - one of which is because this only caches data that is assured to be true *and not change* for a given `CiliumNetworkPolicy` on this node apart from addition / deletion of a rule (i.e., not within the controller for managing the status of the rule). The revision at time of importing the policy as well as whether there was an error importing the given rule will never change in a way that will cause race conditions, because importing the policy / creation of the resultant policy import error are performed synchronously outside of the controller to sync the `CiliumNetworkPolicyNodeStatus`. See my explanation here 
https://github.com/cilium/cilium/pull/5192#discussion_r210093293 for more context around why I did it this way. I did not want to add logic for updating this from the controller (which updates the node status) because the controller runs asynchronously, and calls to update the annotations may occur while the controller is running, which may result in an out-of-date `CiliumNetworkPolicyNodeStatus` being cached locally. 

TL;DR: I viewed this as the safest way of fixing this issue in the context of the 1.2 release due to the fact that these two attributes (revision, import error) are created outside of the controller in a synchronous matter, as opposed to the `CiliumNetworkPolicyNodeStatus` itself.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Kubernetes upstream test failed due to timeout (we need to increase the timeout in the Jenkinsfile):
https://jenkins.cilium.io/job/Cilium-PR-Kubernetes-Upstream/479

Ginkgo tests failed due to a known Kafka test flake, which is unrelated to my changes.: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5687

Flake is tracked by #5204 . I'll run tests again for the sake of doing so.

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
